webapp/repos.py

```diff
@@ -87,24 +87,29 @@ def process_list(self, api_version, data):  # pylint: disable=unused-argument
                 repo_details[label] = self.cache.repo_detail[repo_id]
         filters.append((filter_item_if_exists, [repo_details]))
 
+        actual_page_size = 0
         repo_page_to_process, pagination_response = paginate(repos, page, page_size, filters=filters)
         for label in repo_page_to_process:
             for repo_id in self.cache.repolabel2ids.get(label, []):
                 repo_detail = self.cache.repo_detail[repo_id]
-
-                repolist.setdefault(label, []).append({
-                    "label": label,
-                    "name": repo_detail[REPO_NAME],
-                    "url": repo_detail[REPO_URL],
-                    "basearch": none2empty(repo_detail[REPO_BASEARCH]),
-                    "releasever": none2empty(repo_detail[REPO_RELEASEVER]),
-                    "product": repo_detail[REPO_PRODUCT],
-                    "revision": repo_detail[REPO_REVISION]
-                })
+                if not modified_since_dt or (repo_detail[REPO_REVISION] != 'None' and parse_datetime(
```

Comment:
I would rather write `if not modified_since_dt or self._modified_since(repo_detail, modified_since_dt): ...`.